client/gutenboarding/index.ts

```diff
@@ -9,6 +9,29 @@ import page from 'page';
 import { hideMasterbar, main, redirectIfNotEnabled } from './controller';
 import { makeLayout, render as clientRender } from 'controller';
 
+// Don't complain about window.wp.data types in our debug function
+declare const window: any;
+
 export default function() {
-	page( '/gutenboarding', redirectIfNotEnabled, hideMasterbar, main, makeLayout, clientRender );
+	page(
+		'/gutenboarding',
+		redirectIfNotEnabled,
+		( context, next ) => {
+			if ( process.env.NODE_ENV !== 'production' ) {
+				if ( typeof window === 'object' ) {
+					if ( ! window.wp ) {
+						window.wp = {};
+					}
+					if ( ! window.wp.data ) {
+						window.wp.data = require( '@wordpress/data' );
+					}
+				}
+			}
+			next();
+		},
```

Comment:
Maybe move the middleware def to `./controller`? :sweat_smile:

Reply:
I want to avoid polluting a module with this:

https://github.com/Automattic/wp-calypso/blob/90a22e1c4171ff863b8458bbdc6c513813ce200b/client/gutenboarding/devtools.ts#L6